.github/workflows/e2e.yaml

```diff
@@ -143,79 +152,83 @@ jobs:
-    #     # .net.
-    #     # CoreDNS should handle those domains and answer with NXDOMAIN instead of SERVFAIL
-    #     # otherwise pods stops trying to resolve the domain.
-    #     # Get the current config
-    #     original_coredns=$(/usr/local/bin/kubectl get -oyaml -n=kube-system configmap/coredns)
-    #     echo "Original CoreDNS config:"
-    #     echo "${original_coredns}"
-    #     # Patch it
-    #     fixed_coredns=$(
-    #       printf '%s' "${original_coredns}" | sed \
-    #         -e 's/^.*kubernetes cluster\.local/& net/' \
-    #         -e '/^.*upstream$/d' \
-    #         -e '/^.*fallthrough.*$/d' \
-    #         -e '/^.*forward . \/etc\/resolv.conf$/d' \
-    #         -e '/^.*loop$/d' \
-    #     )
-    #     echo "Patched CoreDNS config:"
-    #     echo "${fixed_coredns}"
-    #     printf '%s' "${fixed_coredns}" | /usr/local/bin/kubectl apply -f -
-
-    # - name: Run tests
-    #   run: |
-    #     export KUBERNETES_CONFORMANCE_TEST='y'
-    #     export E2E_REPORT_DIR=${PWD}/_artifacts
-
-    #     # Run tests
-    #     /usr/local/bin/ginkgo --nodes=25                \
-    #       --focus="\[Conformance\]"     \
-    #       --skip="Feature|Federation|machinery|PerformanceDNS|DualStack|Disruptive|Serial|Slow|KubeProxy|LoadBalancer|GCE|Netpol|NetworkPolicy|NodeConformance"   \
-    #       /usr/local/bin/e2e.test                       \
-    #       --                                            \
-    #       --kubeconfig=${PWD}/_artifacts/kubeconfig.conf     \
-    #       --provider=local                              \
-    #       --dump-logs-on-failure=false                  \
-    #       --report-dir=${E2E_REPORT_DIR}                \
-    #       --disable-log-dump=true
-
-    # - name: Upload Junit Reports
-    #   if: always()
-    #   uses: actions/upload-artifact@v2
-    #   with:
-    #     name: kind-junit-${{ env.JOB_NAME }}-${{ github.run_id }}
-    #     path: './_artifacts/*.xml'
-
-    # - name: Export logs
-    #   if: always()
-    #   run: |
-    #     /usr/local/bin/kind export logs --name ${KIND_CLUSTER_NAME} --loglevel=debug ./_artifacts/logs
-
-    # - name: Upload logs
-    #   if: always()
-    #   uses: actions/upload-artifact@v2
-    #   with:
-    #     name: kind-logs-${{ env.JOB_NAME }}-${{ github.run_id }}
-    #     path: ./_artifacts/logs
+    - name: Get Cluster status
+      run: |
+        # wait network is ready
+        sleep 5
+        /usr/local/bin/kubectl get nodes -o wide
+        /usr/local/bin/kubectl get pods -A
+        /usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods --namespace=kube-system -l k8s-app=kube-dns
+        # smoke test
+        /usr/local/bin/kubectl run test --image httpd:2
+        /usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods test
+        /usr/local/bin/kubectl get pods -A -owide
+        if [ ${{ matrix.ipFamily }} = "ipv4" ];then 
+          sleep 50s
```

Comment:
no sleeps, use `/usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods --namespace=kube-system -l ` and pick a label or somethign that selects the agent

Reply:
Actually, all of pod is running :( and the svc is fine. But the ANP agent complained that svc could not be found.

Reply:
The second temporary solution I can think of is to check the metrics of the ANP server. If  `connectivity_network_proxy_server_grpc_connections` = the size of ANP agent, then continue.

But as a temporary solution, I don’t think it’s worth investing in.

Reply:
have you tried adding a readiness probe  on the daemonset/pods?

Reply:
Thanks for your remind, make sense for me, We no longer need sleep now.

I didn't try readiness at the beginning because I remembered that the ANP agent had no implementation that could check if the server was connected, until I found it. https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/485

Tested it 16 times and it's working.